lib/code-ribbon-ribbon-container.js

```diff
@@ -135,28 +152,89 @@ class CodeRibbonRibbonContainer extends PaneAxis {
       n_crsr.initialize();
     }
 
+    // increment / remove patchQuota
+    if (
+      this.autoincrement_patch_quota() == false
+    ) {
+      this.autoremove_excess_empty_right_patch_column();
+    }
+
     // update all the children
     this.children.map(child => {
       child.cr_update();
     });
 
     // set width and scrolling etc,
-    // find the max number of items of any child:
-    this.max_horiz_patches = 1;
-    // crdebug("Finding max_horiz_patches...")
+    var widthpercent = 100.0 * (
+      this.patchQuota / atom.config.get("code-ribbon.pane_count_calc.pane_count_horizontal_number")
+    );
+    this.getElement().style.width = widthpercent.toString() + "%";
```

Comment:
So is this the width of what the user will be able to see?

Reply:
This is the HTML width of the ribbon container, which is most often many times wider than the user's actual screen.